go/kbfs/libkbfs/leveldb.go

```diff
@@ -35,19 +43,87 @@ var leveldbOptions = &opt.Options{
 	// X, and >=1024 on (most?) Linux machines. So set to a low
 	// number since we have multiple leveldb instances.
 	OpenFilesCacheCapacity: 10,
+	CompactionTotalSize:    15 * opt.MiB,
+	CompactionL0Trigger:    6,
 }
 
 // LevelDb is a libkbfs wrapper for leveldb.DB.
 type LevelDb struct {
 	*leveldb.DB
-	closer io.Closer
+	storageCloser io.Closer
+	options       *opt.Options
+
+	compactionLock   sync.Mutex
+	compactionQueued bool
+	compactionTime   time.Time
+	compactionTimer  *time.Timer
+	shutdownC        chan struct{}
+}
+
+// resets the compaction timer. When the compaction timer hits 0,
+// the DB considers compaction more strongly.
+func (ldb *LevelDb) resetTimer() {
+	ldb.compactionLock.Lock()
+	defer ldb.compactionLock.Unlock()
+	ldb.compactionTime = time.Now().Add(compactionTimeout)
```

Comment:
Should probably be using a `clockwork.Clock` (https://godoc.org/github.com/jonboulle/clockwork#Clock), so you can simulate pauses in tests.